pkg/apis/pingcap/v1alpha1/tidbcluster.go

```diff
@@ -262,6 +262,9 @@ func (tc *TidbCluster) PDAllPodsStarted() bool {
 }
 
 func (tc *TidbCluster) PDAllMembersReady() bool {
+	if tc.Spec.PD == nil {
```

Comment:
What about checking `tc.spec.pd !=nil` before calling this function?
Similar with the following change for TiKV and TiDB.
cc @cofyc